pkg/workloads/containerd/watcher.go

```diff
@@ -66,7 +99,7 @@ func EnableEventListener() error {
 		return err
 	}
 
-	go listenForDockerEvents(r)
+	go listenForDockerEvents(&eventQueue, r)
```

Comment:
Given that we only enqueue now, we want to run this in sync otherwise the enqueue itself could get out of order.

Reply:
I'm not sure I follow you. We still have a race between syncWithRuntime and listenForDockerEvents. listenForDockerEvents is annoying to call from syncWithRuntime because it spends it's time reading a Reader. As you suggested elsewhere, though, we can pass a "since" option to the queries for both events and containerList. I haven't touched that here, though, but it's the next thing I was going to do.
I'm happy to have it all in one PR.

Reply:
Forget what I said, I misread the code. I thought that was invoked per event. This is fine.